devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -657,7 +657,12 @@ void prepare(final boolean triggerCompile) throws Exception {
             //in most cases these are not used, however they need to be present for some
             //parent-first cases such as logging
             for (Artifact appDep : project.getArtifacts()) {
-                addToClassPaths(classPathManifest, appDep.getFile());
+                // only add the artifact if it's present in the dev mode context
+                // we need this to avoid having jars on the classpath multiple times
+                if (!devModeContext.getLocalArtifacts().contains(new AppArtifactKey(appDep.getGroupId(), appDep.getArtifactId(),
+                        appDep.getClassifier(), appDep.getType()))) {
```

Comment:
appDep.getArtifactHandler().getExtension() perhaps would be a more accurate one.

Reply:
Didn't know that one, thanks!